services/api/src/resources/project/harborSetup.js

```diff
@@ -24,133 +25,293 @@ const lagoonWebhookAddress = R.compose(
+      logger.info(`Unable to create the harbor project "${lagoonProjectName}", as it already exists in harbor!`)
+    } else {
+      logger.error(`Unable to create the harbor project "${lagoonProjectName}", error: ${err}`)
+      return ""
+    }
+  }
+
+  // Get new harbor project's id
+  try {
+    const res = await harborClient.get(`projects?name=${lagoonProjectName}`)
+    var harborProjectID = res.body[0].project_id
+    logger.debug(`Harbor project id for ${lagoonProjectName} is: ${harborProjectID}`)
+  } catch (err) {
+    if (err.statusCode == 404) {
+      logger.error(`Unable to get the harbor project id of "${lagoonProjectName}", as it does not exist in harbor!`)
+      return
+    } else {
+      logger.error(`Unable to get the harbor project id of "${lagoonProjectName}" !!`)
+      return ""
+    }
+  }
+  return harborProjectID
+}
+
+async function createRobot(sqlClient /* : MariaSQL */, harborClient /* : harborClient */, lagoonProjectName /* : string */, harborProjectID /* : string */, robot /* : string */) {
+  // Returns false on an error and a token object on a success
+
+  // Create robot account for new harbor project
+  try {
+    const timestamp = Math.floor(Date.now() / 1000)
+    var robotName = (robot != "") ? robot : `${lagoonProjectName}-${timestamp}`
+
+    const res = await harborClient.post(`projects/${harborProjectID}/robots`, {
+      body: {
+        name: robotName,
+        access: [
+          {
+            resource: `/project/${harborProjectID}/repository`,
+            action: "push"
+          }
+        ]
       }
+    })
+    var harborTokenInfo = res.body
+    logger.debug(`Robot ${robotName} was created for Harbor project ${lagoonProjectName} !`)
+  } catch (err) {
+    // 409 means project already exists
+    // 201 means project created successfully
+    if (err.statusCode == 409) {
+      logger.warn(`Unable to create a robot account for harbor project "${lagoonProjectName}", as a robot account of the same name already exists!`)
+    } else {
+      logger.error(`Unable to create a robot account for harbor project "${lagoonProjectName}"!`, err)
+      return false
     }
+  }
+  return harborTokenInfo
+}
 
-    // Get new harbor project's id
-    try {
-      const res = await harborClient.get(`projects?name=${lagoonProjectName}`)
-      var harborProjectID = res.body[0].project_id
-      logger.debug(`Got the harbor project id for project ${lagoonProjectName} successfully!`)
-    } catch (err) {
-      if (err.statusCode == 404) {
-        logger.error(`Unable to get the harbor project id of "${lagoonProjectName}", as it does not exist in harbor!`)
-      } else {
-        logger.error(`Unable to get the harbor project id of "${lagoonProjectName}" !!`)
+async function removeRobot(sqlClient /* : MariaSQL */, harborClient /* : harborClient */, lagoonProjectName /* : string */, harborProjectID /* : string */, robotName /* : string */) {
```

Comment:
this code is never used? I guess it can be removed then?

Reply:
Eventually, we will need to handle the removal of older robot accounts; this function will be relevant when we want to implement that removal functionality.

Reply:
okay, but then please let's not ship code that is not used anywhere, remove the code from this PR, and add it to another branch that we can then use again when the time is right for the removal functionality.

Reply:
Understood. Function removed and stored in the `add_harbor_robot_removal` branch.